pkg/trigger/trigger.go

```diff
@@ -0,0 +1,65 @@
+package trigger
+
+import (
+	"sync"
+)
+
+type Trigger interface {
+	Kick(id string) chan struct{}
+	Triggered() <-chan struct{}
+	Ready(id string)
+}
```

Comment:
Non-blocking design thoughts:

What things would look like if our agents processed jobs using a single "workqueue" interface, like [the one used by k8s controllers](https://github.com/kubernetes/kubernetes/blob/227c2e7c2b2c05a9c8b2885460e28e4da25cf558/staging/src/k8s.io/client-go/examples/workqueue/main.go#L40)?

I haven't ruminated too deeply on this, but my naive intuition is that a workqueue would let us share a lot more code and swap out the underlying messaging mechanism (e.g. polling sqlite, channels, message broker, etc); the end goal being the boiling down of agent implementations to a few key methods, like `Run(context.Context, Request) (Response, error)`.

@thedadams, considering your recent foray into mink/baaah for clicky-chats, do you have any thoughts on this approach?

Reply:
Donnie and I spoke about this outside of GitHub.